### PR TITLE
moved is_inbound_allowed to NetworkState

### PR DESCRIPTION
**Pull request by pompon0**
part of the larger effort to move all functionality from PeerManagerActor to NetworkState.

### CODE REVIEW

File: chain/network/src/peer_manager/network_state/mod.rs [hunk 1]
```diff
@@ -34,6 +36,28 @@ pub(crate) const LIMIT_PENDING_PEERS: usize = 60;
...
+            id: pi.id.clone(),
+            addr: if let Some(addr) = pi.addr {
+                addr.clone()
+            } else {
+                anyhow::bail!("addess is missing");
+            },
+            account_id: pi.account_id.clone(),
+        })
+    }
+}
```
Comment (matklad) [inline chain/network/src/peer_manager/network_state/mod.rs:52]: drive by, but `TryFrom<&T>` like these look pretty ugly. TryFrom/From is usually about ownership transport or format changes, it's not a generall one-argument constructor. So, something like

```
impl WhitelistNode {
  pub fn from_peer_info(pi: &PeerInfo) -> anyhow::Result<Self>
}
```

would read better. 

The most immediate benefit is that its far easier to find all call-sites.

File: chain/network/src/peer_manager/network_state/mod.rs [hunk 2]
```diff
@@ -70,6 +94,16 @@ pub(crate) struct NetworkState {
...
+
+    /// Whitelisted nodes, which are allowed to connect even if the connection limit has been
+    /// reached.
+    whitelist_nodes: Vec<WhitelistNode>,
+    /// Maximal allowed number of peer connections.
+    /// It is initialized with config.max_num_peers and is mutable
+    /// only so that it can be changed in tests.
+    /// TODO(gprusak): determine why tests need to change that dynamically
+    /// in the first place.
+    pub max_num_peers: AtomicCell<u32>,
```
Comment (matklad) [inline chain/network/src/peer_manager/network_state/mod.rs:106]:
```suggestion
    pub max_num_peers: AtomicU32,
```

**Review by matklad — APPROVED** [review]

File: chain/network/src/peer_manager/network_state/mod.rs (hunk 2 above)
Comment (pompon0) [reply chain/network/src/peer_manager/network_state/mod.rs:106]: done

File: chain/network/src/peer_manager/network_state/mod.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/peer_manager/network_state/mod.rs:52]: done

**Pull request merged**